src/colony/modules/net.js

```diff
@@ -163,7 +147,17 @@ TCPSocket.prototype.connect = function (/*options | [port], [host], [cb]*/) {
 
   function setUpConnection(ip) {
     if (self.socket == null) {
-      if (self._secure) ensureSSLCtx();
+      if (self._secure) {
+        var custom_certs = null;
+        self._ssl_checkCerts = (opts.rejectUnauthorized !== false);
```

Comment:
To be clear, (for this whole function) is `opts` ever optional? Here it's assumed non-null.